strategies/categorized/src/main/java/org/arquillian/smart/testing/strategies/categorized/TagsParser.java

```diff
@@ -0,0 +1,127 @@
+package org.arquillian.smart.testing.strategies.categorized;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.arquillian.smart.testing.logger.Log;
+import org.arquillian.smart.testing.logger.Logger;
+
+class TagsParser {
+
+    public static final String TAG = "org.junit.jupiter.api.Tag";
+    public static final String TAGS = "org.junit.jupiter.api.Tags";
+    private final Logger logger = Log.getLogger();
+    private final CategorizedConfiguration strategyConfig;
+    private final List<String> specifiedCategories;
+
+    TagsParser(CategorizedConfiguration strategyConfig) {
+        this.strategyConfig = strategyConfig;
+        specifiedCategories = Arrays.stream(strategyConfig.getCategories())
+            .map(this::changeIfNonCaseSensitive)
+            .collect(Collectors.toList());
+    }
```

Comment:
How about creating a new abstract class `Parser` with the common logic??

Reply:
The method `hasCorrectTags` is pretty similar to `hasCorrectCategories` - that could be inside of the abstract class as well.